### ISSUE TITLE
How to run with a list of point cloud.

### ISSUE THREAD

**Issue by AbyssGaze** [sdmiller/cpu_tsdf#23]
I follow the readme file to construct tsdf from a list of point cloud, but the result is null.

```
#include <cpu_tsdf/tsdf_interface.h>
#include <cpu_tsdf/tsdf_volume_octree.h>
#include <cpu_tsdf/marching_cubes_tsdf_octree.h>

#include <pcl/console/print.h>
#include <pcl/io/ply_io.h>
#include <pcl/io/pcd_io.h>

#include <string>
#include <vector>

using namespace std;
using namespace cpu_tsdf;
using namespace pcl;

int main (int argc, char** argv)
{
    if (argc < 3)
    {
        PCL_INFO ("This is a utility program meant to render a mesh from a TSDF Volume, which can be saved to disk via TSDFVolumeOctree::save(const std::string &filename).\n");
        PCL_INFO ("Usage: %s  foo.ply pcd_file.txt", argv[0]);
        return (1);
    }
    string mesh_file = argv[1];

    TSDFVolumeOctree::Ptr tsdf (new TSDFVolumeOctree);
    tsdf->setGridSize (0.5, 0.5, 0.5); 
    tsdf->setResolution (128, 128, 128); 

    tsdf->setIntegrateColor (true); 
    Eigen::Affine3d tsdf_center;
    tsdf->setGlobalTransform (tsdf_center);
    tsdf->reset ();

    ifstream pcdFile(argv[2], ifstream::in);
    string pcdFileName;
    while(pcdFile >> pcdFileName)
    {
        PointCloud<pcl::PointXYZRGB>::Ptr cloud(new PointCloud<PointXYZRGB>);
        if (io::loadPCDFile <PointXYZRGB>(pcdFileName, *cloud) != -1)
        {
            tsdf->integrateCloud (*cloud, PointCloud<Normal> (), Eigen::Affine3d::Identity ()); // Integrate the cloud
        }
    // Note, the normals aren't being used in the default settings. Feel free to pass in an empty cloud
    }
    // Now what do you want to do with it?
    float distance;
    PointXYZ query_point (0, 0, 0);
    tsdf->getFxn (query_point, distance); 
    tsdf->save ("output1.vol");

    // Mesh with marching cubes
    MarchingCubesTSDFOctree mc;
    mc.setInputTSDF (tsdf);
    mc.setMinWeight (2);
    mc.setColorByRGB (true); 
    PolygonMesh mesh;
    mc.reconstruct (mesh);
    cout << mesh.cloud.data.size() << endl;
    io::savePLYFileBinary (mesh_file, mesh);

}
```
I had convert the pcd file to the same view that the pose is Eigen::Affine3d::Identity () , and the cloud is small for reconstructing object that I set **setGridSize (0.5, 0.5, 0.5)**. But the after reconstruct the result of mesh is null, I don't know how to revise.
Thank you!
@sdmiller

**Comment by sdmiller** [comment]
Hi @AbyssGaze,

At first glance I don't see anything particularly wrong with this. How many clouds are you feeding in? I wonder if the MinWeight of 2 is hurting things (though still, purely empty is surprising.)

Feel free to e-mail me at sdavidmiller __AT__ gmail -- might be easier to send some sample data and get the ball rolling quicker.

**Comment by sdmiller** [comment]
Update: we've talked a bit internally, and I've tried running on her data. So far the key issue (which I should probably have big and bold in the README) is that the clouds weren't organized. Since the TSDF algorithm hinges on raytracing, any operation which removes our ability to correlate a point with the pixel it came from will result in a broken (or, in this case, empty) result.

Closing this for now, will open and update as we find more things. I'll be shocked if I don't have a few bugs still hiding somewhere on my end :).
